Added empty podcasts screen in watch app

atharv028 opened the pull request:
## Description

This PR adds a message screen for wear app when the user isn't subscribed to any podcasts, showing message to subscribe to podcasts on their phone.

Fixes #1076

## Testing Instructions
1. Login to the wear app.
2. Click on podcasts.
3. ✅ Verify the subscribe screen is shown when the user doesn't have any subscribed podcasts, telling them to subscribe to podcast on phone
4. ✅ Verify that the previous flow when the user does have subscribed podcast(s) works.

## Screenshots or Screencast 
<img src="https://github.com/Automattic/pocket-casts-android/assets/89896473/41c571ed-67dd-4494-84b1-0fe1975bed5d" width="40%" height="40%">

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
 
#### I have tested any UI changes...
<!-- If this PR does not contain UI changes, ignore these items -->
- [x] with the device set to have a large display and font size

ashiagr commented on wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/ui/podcasts/PodcastsScreen.kt:
```diff
@@ -52,15 +56,34 @@ fun PodcastsScreen(
-                }
-                is FolderItem.Folder -> {
-                    FolderChip(folderItem = item, onClick = navigateToFolder)
+        if (uiState.items.isNotEmpty()) {
```
This is good, it might be slightly better if we move this logic to the ViewModel and handle `Empty`, `Loaded` states returned from it in the `PodcastScreen`. We can convert `PodcastsViewModel` `UiState` to a sealed class as below:

```
sealed class UiState {
    object Empty : UiState()
    data class Loaded(
         val folder: Folder? = null,
         val items: List<FolderItem> = emptyList()
    ) : UiState()
}
```

ashiagr reviewed (commented):
This is great! Thank you @atharv028 

I left a few minor comments. Once those are addressed, we're ready to merge it.

atharv028 commented:
I have added the ui states class and moved the logic to ViewModel. 

I was thinking about adding a third state `object Loading: UiState()` and set it as the initial state, but I thought that it would be redundant since we are not showing any loader on the watch.

What do you think? Should I add that state and a loader or this one looks good?

<img src="https://github.com/Automattic/pocket-casts-android/assets/89896473/699b5528-7158-4f06-b892-fc77b4d303b0" width="40%" height="40%">

ashiagr commented:
> I was thinking about adding a third state object Loading: UiState() and set it as the initial state, but I thought that it would be redundant since we are not showing any loader on the watch.

Good idea. It makes sense to add it for this screen to give a visual cue that podcasts are loading instead of showing this empty state message. There is a `LoadingSpinner` component that we can display for this state.

ashiagr commented on wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/ui/podcasts/PodcastsScreen.kt:
```diff
@@ -52,15 +56,34 @@ fun PodcastsScreen(
             }
+        } else {
+            item {
+                TextH30(
```
Updated in 06e53b7

atharv028 commented:
Added a `object Loading: UiState()` state to ViewModel and used `LoadingSpinner()` to show the loader.

https://github.com/Automattic/pocket-casts-android/assets/89896473/9f8f1563-5e10-4108-8637-1dded22e8cb2

ashiagr reviewed (approved):
Looks good! Thanks for the updates. 🙇‍♀️
